File: pilhas/pilhas.c
#include <stdio.h>
#include <stdlib.h>

#define MAX 5  // Tamanho máximo da pilha

typedef struct {
    int topo;
    int itens[MAX];
} Stack;

// Inicializa a pilha
void init(Stack *s) {
    s->topo = -1;
}

// Verifica se a pilha está vazia
int isEmpty(Stack *s) {
    return s->topo == -1;
}

// Adiciona elemento na pilha
void push(Stack *s, int item) {
    if (s->topo == MAX - 1) {
        printf("A pilha está cheia!\n");
        return;
    }
    s->itens[++s->topo] = item;
}

// Remove elemento da pilha
int pop(Stack *s) {
    if (isEmpty(s)) {
        printf("A pilha está vazia!\n");
        return -1;
    }
    return s->itens[s->topo--];
}

// Mostra o topo da pilha
int peek(Stack *s) {
    if (isEmpty(s)) {
        return -1;
    }
    return s->itens[s->topo];
}

// Testando a pilha em C
int main() {
    Stack s;
    init(&s);

    push(&s, 10);
    push(&s, 20);
    push(&s, 30);

    printf("Topo: %d\n", peek(&s)); // Saída: 30
    printf("Removido: %d\n", pop(&s)); // Saída: 30
    printf("Topo: %d\n", peek(&s)); // Saída: 20

    return 0;
}

File: pilhas/pilhas.py
class Stack:
    def __init__(self):
        self.items = []

    def push(self, item):
        self.items.append(item)  # Insere no topo

    def pop(self):
        if not self.is_empty():
            return self.items.pop()  # Remove do topo
        return "A pilha está vazia!"

    def peek(self):
        if not self.is_empty():
            return self.items[-1]  # Último elemento
        return None

    def is_empty(self):
        return len(self.items) == 0

    def size(self):
        return len(self.items)

stack = Stack()
stack.push(10)
stack.push(20)
stack.push(30)
print(stack.pop()) 
print(stack.peek())  
